addons/zylann.hterrain/util/xyz_size_calc.gd

```diff
@@ -0,0 +1,37 @@
+class_name XYZMeta
+
+const MAX_INT = 9223372036854775807 # 2^63 - 1
+
+var size_x = -1
+var size_y = -1
+var min_x = MAX_INT
+var min_y = MAX_INT
+var max_x = 0
+var max_y = 0
+
+
+func get_XYZ_Meta(f, logger):
+	if size_x < 0 or size_y < 0:
```

Comment:
No idea why that `if` exists. Seems like you expect a race condition, which doesnt sound like a good idea in a utility function. In general that's why `static func` is preferable because they are better self-contained and don't assume things to happen before the call.

Reply:
I planned on doing a singleton so I could "cache" the result, but singletons don't reall exist in GDScript it seems.